pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/VarianceAggregationFunction.java

```diff
@@ -116,8 +139,20 @@ protected void setGroupByResult(int groupKey, GroupByResultHolder groupByResultH
   public void aggregateGroupBySV(int length, int[] groupKeyArray, GroupByResultHolder groupByResultHolder,
       Map<ExpressionContext, BlockValSet> blockValSetMap) {
     double[] values = StatisticalAggregationFunctionUtils.getValSet(blockValSetMap, _expression);
-    for (int i = 0; i < length; i++) {
-      setGroupByResult(groupKeyArray[i], groupByResultHolder, 1L, values[i], 0.0);
+    RoaringBitmap nullBitmap = null;
+    if (_nullHandlingEnabled) {
+      nullBitmap = blockValSetMap.get(_expression).getNullBitmap();
+    }
+    if (nullBitmap != null && !nullBitmap.isEmpty()) {
+      for (int i = 0; i < length; i++) {
+        if (!nullBitmap.contains(i)) {
+          setGroupByResult(groupKeyArray[i], groupByResultHolder, 1L, values[i], 0.0);
+        }
+      }
+    } else {
+      for (int i = 0; i < length; i++) {
+        setGroupByResult(groupKeyArray[i], groupByResultHolder, 1L, values[i], 0.0);
+      }
```

Comment:
We need to also add similar logic to `aggregateGroupByMV`. We don't support MV as null right now, but we should support regular MV group key